src/plugins/intel_npu/src/al/include/intel_npu/al/config/config.hpp

```diff
@@ -373,6 +415,25 @@ void OptionsDesc::add() {
     }
 }
 
+template <class Opt>
+void OptionsDesc::add(compilerVersion compilerVersionRequirement) {
+    if (compilerVersionRequirement.vclMajor >= Opt::compilerSupportVersion().vclMajor &&
```

Comment:
I assume we use this check to replace the option compatibility check in zero adapter. If we support an option from 5.7 and drop from 5.12, how to support such context?

User application may still use old configs, once they use new npu plugin, it can not work with old driver. I think we need to remove this issue for user.

Reply:
Yes, this has 2 main purposes:
1. get rid of the compatibility checks in zero adapter
2. enables us register properties dynamically, so we will broadcast only the truly supported properties.

The case of when an option gets dropped in the future is indeed not covered. I am working on a redesign on this part to enable this.